[mkdocs] docs: add faucet tutorial (#1759)
symbol/symbol: mkdocs/pages/en/devbook/accounts/testnet-faucet.md

---
title: Fund via Faucet
---

# Getting Testnet Funds from the Faucet

The Symbol testnet provides a faucet that distributes free <XYM:> to developer <accounts:> for testing purposes.
This guide explains how to claim testnet funds using the web-based faucet.

!!! note
    Testnet XYM has no real-world value.
    It exists only to let you experiment with Symbol features without using real currency.

    If you need mainnet XYM, see available

>     [exchanges](https://coinmarketcap.com/currencies/symbol/#Markets).

Reviewer: ```suggestion
    If you need mainnet XYM, you will need to buy it through an
    [exchange](https://coinmarketcap.com/currencies/symbol/#Markets).
```

## Prerequisites

Before you start, make sure to:

- Create a testnet <account:> to receive funds.

>   See [Creating an Account from a Private Key](../accounts/create-from-private-key.md).

Reviewer: ```suggestion
- Create a testnet <account:> to receive funds, either
  [from code](../accounts/create-from-private-key.md) or
  [by using a wallet](../../userbook/wallet/create-account.md).
```

> - Have an X account to verify your identity with the faucet.

Reviewer: ```suggestion
- Have an 𝕏 account to verify your identity with the faucet.
```

## How to Claim Testnet Funds

{% import 'tutorial.jinja2' as tutorial %}

{{ tutorial.list_begin() }}

{{ tutorial.step_begin("../../images/faucet-open.png") }}
Open your web browser and navigate to the Symbol testnet faucet at

> **<https://testnet.symbol.tools/>**.

Reviewer: ```suggestion
[testnet.symbol.tools](https://testnet.symbol.tools).
```

The raw URL wasn't rendering, for some reason.

{{ tutorial.step_end() }}

{{ tutorial.step_begin("../../images/faucet-sign-in.png") }}

> Click **Sign in with Twitter** (now X) and follow the authentication flow.

Reviewer: ```suggestion
Click **Sign in with Twitter** (now 𝕏) and follow the authentication flow.

This step limits the amount of test funds to 10,000 XYM per account,
to help prevent abuse of the faucet.
```

{{ tutorial.step_end() }}

{{ tutorial.step_begin("../../images/faucet-authorize.png") }}

> After signing in, X will ask you to authorize the faucet application to access

Reviewer: ```suggestion
After signing in, 𝕏 will ask you to authorize the faucet application to access
```

your account information.

Review the permissions and click **Authorize app** to continue.
Once authorized, you will be redirected again to the faucet.
{{ tutorial.step_end() }}

> {{ tutorial.step_begin("../../images/faucet-address.png") }}

Reviewer: The picture does not show any field called "Your Testnet Address", so it's confusing.
I'd rather show the picture _before_ you enter the address, so the placeholder text is visible.
Also, if you hover over the field before taking the screenshot, it gets a nice purple glow that will highlight the field even more.

Enter your testnet address in **Your Testnet Address** field.

> The address should start with `T` for testnet accounts.

Reviewer: ```suggestion
Enter the address where you want to receive the funds in the **Your Testnet Address** field.

Make sure the address starts with `T`, meaning it's a testnet account.
```

{{ tutorial.step_end() }}

> {{ tutorial.step_begin("../../images/faucet-xym.png") }}

Reviewer: Same comment as before, the picture does not show any field called "XYM Amount".

In the **XYM Amount** field, specify how much XYM you want to claim.
The maximum amount per request is 10,000 XYM.
{{ tutorial.step_end() }}

{{ tutorial.step_begin("../../images/faucet-claim.png") }}
Click **Claim** to submit your request.
If the request is successful, the faucet will transfer the specified amount of XYM to your address.
{{ tutorial.step_end() }}

{{ tutorial.step_begin("../../images/faucet-view-explorer.png") }}
Click **View in Explorer** in the top-right corner notification to verify that the transaction was processed.

> The explorer will display the transaction details, including its confirmation status.

Reviewer: Consider adding a sentence about how confirmation is expected to last 30 seconds at most, and that the transfer can also be monitored from the wallet.

Author: Added a note about the wallet. I preferred not to include timing details, since confirmation can take longer than 30 seconds (e.g. if the block is full). Instead, users should wait for the confirmation status or verify that the XYM appears in their account.

Reviewer: I like to give at least an order of magnitude. I've seen faucets that need human approval, so they can take days before you receive the funds.
Maybe we don't need to say "30 seconds" but we can say "a few seconds" or "less than a minute" so readers know what to expect.

Author: Addressed in https://github.com/symbol/symbol/pull/1759/commits/32ff84af6dac46d36792725f1041c9b46f3be25a
